Add "Colors" extension

linuz90 opened the pull request:
## Description

This extension uses the `tinycolor` library to convert your provided color into various formats. It's very handy for designers and developers that need to do quick conversions.

For example, I often prefer HSL to RGB or HEX since it lets me change saturation and luminance more easily.

You can press enter on a converted color to copy it, or Command + Enter to insert it inline.

## Type of change

- New extension

## Screencast

https://user-images.githubusercontent.com/9629188/151674382-e431c38a-204b-4d36-b2ab-045cf14b6979.mp4

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

linuz90 commented:
Forgot to showcase that invalid colors will throw an error:

![CleanShot 2022-01-29 at 19 17 45@2x](https://user-images.githubusercontent.com/9629188/151674542-87e715ff-a6ac-48c4-be7b-4765bf059870.png)

pernielsentikaer commented on extensions/colors/src/index.tsx:
```diff
@@ -0,0 +1,110 @@
+import {
+  ActionPanel,
+  ActionPanelItem,
+  CopyToClipboardAction,
+  Form,
+  Icon,
+  List,
+  PasteAction,
+  SubmitFormAction,
+  Toast,
+  ToastStyle,
+  useNavigation,
+} from "@raycast/api";
```
```suggestion
import {
  ActionPanel,
  ActionPanelItem,
  CopyToClipboardAction,
  Form,
  Icon,
  List,
  PasteAction,
  showToast,
  SubmitFormAction,
  ToastStyle,
  useNavigation,
} from "@raycast/api";
```

pernielsentikaer commented on extensions/colors/src/index.tsx:
```diff
@@ -0,0 +1,110 @@
+
+  function handleSubmit(form: { color: string }) {
+    const color = form.color;
+
+    const toast = new Toast({ title: "", style: ToastStyle.Failure });
+    toast.hide();
+
```
```suggestion
```

pernielsentikaer commented on extensions/colors/src/index.tsx:
```diff
@@ -0,0 +1,110 @@
+    toast.hide();
+
+    if (!color) {
+      toast.title = "Please enter a color";
+      toast.show();
```
```suggestion
      showToast(ToastStyle.Failure, "Please enter a color");
```

pernielsentikaer commented on extensions/colors/src/index.tsx:
```diff
@@ -0,0 +1,110 @@
+    }
+    const parsedColor = tinycolor(color);
+    if (!parsedColor.isValid()) {
+      toast.title = "This is not a valid color";
+      toast.show();
```
```suggestion
      showToast(ToastStyle.Failure, "This is not a valid color");
```

pernielsentikaer commented on extensions/colors/src/index.tsx:
```diff
@@ -0,0 +1,110 @@
+
+  return (
+    <Form
+      navigationTitle="Insert color to convert"
```
```suggestion
      navigationTitle="Insert Color to Convert"
```

pernielsentikaer commented on extensions/colors/src/index.tsx:
```diff
@@ -0,0 +1,110 @@
+  const c = parsed.toHexString();
+
+  return (
+    <List navigationTitle="Converted colors">
```
```suggestion
    <List>
```
Redundant

pernielsentikaer commented on extensions/colors/src/index.tsx:
```diff
@@ -0,0 +1,110 @@
+    <List navigationTitle="Converted colors">
+      <List.Section>
+        <List.Item
+          title="Convert new color"
```
```suggestion
          title="Convert New Color"
```

pernielsentikaer commented on extensions/colors/src/index.tsx:
```diff
@@ -0,0 +1,110 @@
+          icon={Icon.ArrowClockwise}
+          actions={
+            <ActionPanel>
+              <ActionPanelItem title="New color" onAction={() => navigation.pop()} />
```
```suggestion
              <ActionPanelItem title="New Color" onAction={() => navigation.pop()} />
```

pernielsentikaer commented on extensions/colors/src/index.tsx:
```diff
@@ -0,0 +1,110 @@
+          }
+        />
+      </List.Section>
+      <List.Section title="Converted colors">
```
```suggestion
      <List.Section title="Converted Colors">
```

pernielsentikaer commented on extensions/colors/src/index.tsx:
```diff
@@ -0,0 +1,110 @@
+        <ColorItem description="HSL" color={c} value={parsed.toHslString()} />
+        <ColorItem description="HSV" color={c} value={parsed.toHsvString()} />
+        <ColorItem
+          description="Perceived brightness"
```
```suggestion
          description="Perceived Brightness"
```

pernielsentikaer commented on extensions/colors/src/index.tsx:
```diff
@@ -0,0 +1,110 @@
+const ColorActions = ({ content }: { content: string }) => {
+  return (
+    <ActionPanel>
+      <CopyToClipboardAction title="Copy to clipboard" content={content} />
```
```suggestion
      <CopyToClipboardAction title="Copy to Clipboard" content={content} />
```

pernielsentikaer commented on extensions/colors/src/index.tsx:
```diff
@@ -0,0 +1,110 @@
+  return (
+    <ActionPanel>
+      <CopyToClipboardAction title="Copy to clipboard" content={content} />
+      <PasteAction title="Insert" content={content} shortcut={{ modifiers: ["cmd"], key: "enter" }} />
```
```suggestion
      <PasteAction title="Paste" content={content} shortcut={{ modifiers: ["cmd"], key: "enter" }} />
```

pernielsentikaer reviewed (changes requested):
Hi @linuz90 👋 

Thanks for your contribution 💪 

I left some suggestions for you to work with, also, it could be cool with:
- A way to select default action, if copy or paste should be default ([Like done in this extension](https://github.com/raycast/extensions/pull/593/files))

- A new command that takes selection as color value (idea)

- A new command that takes clipboard as color value (idea)

Looking forward to testing this out again 💪 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

fe9lix commented:
Hey @linuz90, what's the status of this PR? Are you planning on addressing the feedback from @pernielsentikaer?

linuz90 commented:
> Hey @linuz90, what's the status of this PR? Are you planning on addressing the feedback from @pernielsentikaer?

Hey, I'd love to but I'm just too busy working on Typefully now, so I'll likely won't have time for this for a few weeks.

fe9lix commented:
Are you fine with closing this for now and then re-opening a new PR with the feedback addressed, once you've found time?

linuz90 commented:
Sure, no problem!

pernielsentikaer commented:
Closing as agreed
